_includes/item/download-buttons.html

```diff
@@ -10,8 +10,9 @@
 {%- endcomment -%}
 <div class="btn-group" role="group" aria-label="Item options">
     {% if page.object_transcript %}<button class="btn btn-outline-primary" type="button" data-bs-toggle="collapse" data-bs-target="#collapseTranscript" aria-expanded="false" aria-controls="collapseExample">View Transcript</button>{% endif %}
-    {% if page.date %}{%- capture year -%}{% if page.date contains "-" %}{{ page.date | split: "-" | first }}{% elsif page.date contains "/" %}{{ page.date | split: "/" | last }}{% else %}{{ page.date }}{% endif %}{%- endcapture -%}
-    <a href="{{ year | strip | prepend: '/timeline.html#y' | relative_url }}" class="btn btn-outline-primary">View on Timeline</a>{% endif %}
+    {% if page.date %}{% assign regex = '[\dX]{4}' %}{%- assign matches = page.date | regex_match: regex -%}
+    {%- capture url -%}{%- for match in matches -%}{% capture queryParam %}y{{ match }}{% endcapture %}{% assign years = years | append: queryParam %}{% assign url = url | append: queryParam %}{%- endfor %}{%- endcapture %}{% assign url = url | append: '#y' | append: matches[0] %}{% assign url = '?' | append: years | append: '#y' | append: matches[0] %}   
```

Comment:
I do not fully understand what happens here. Can you explain this to me on Thursday?

Reply:
I guess I understand now. You only use the query params as a way to save params client side. They are not used/read by the server, right?

Reply:
Exactly.
Here I build the link from the dates of the item. The query params are then used in the script __includes/js/timeline-js.html_ to mark the rows (years) in the timeline that contain the item. I had to adapt this, as an item can be in several rows now (= all years in the edtf-date of the item).

Reply:
Hacky but nice! ;)